### PR TITLE
Adding VIA support for 40percentclub/luddite

### PR DESCRIPTION
**Pull request by morbetter**
## Description

Added rules.mk and VIA keymap.  Also added config.h to override the number of LEDs for underglow.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/40percentclub/luddite/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#undef RGBLED_NUM
```
Comment (zvecr) [inline keyboards/40percentclub/luddite/keymaps/via/config.h]: As discussed on discord, I'm not sure this is the right change here.

Given the board has just a 3pin header which you can solder any number of leds, 16 is just a personal preference vs the existing default of 8. Personally I would like to see the default via keymap also use the boards default value. And if required another `morbetter` keymap added to cover your personal use case.

Comment (morbetter) [reply keyboards/40percentclub/luddite/keymaps/via/config.h]: Would it make more sense to just not have LEDs at all as the default?  The board doesn't include have any pads, and it's entirely up to the user to add whatever quantity of LEDs.

File: keyboards/40percentclub/luddite/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+};
+
```
Comment (fauxpark) [inline keyboards/40percentclub/luddite/keymaps/via/keymap.c]: This block can be removed.

**Comment by morbetter** [comment]
Since the Luddite is just one of several 40percentclub boards, would it make more sense to use this or something similar?

`#define PRODUCT_ID      0x3430 // "40"`

**Comment by fauxpark** [comment]
The vendor ID should be changed too IMO.

**Comment by morbetter** [comment]
It looks like 0x3430 isn't taken yet.  Should I use that?

**Comment by noroadsleft** [comment]
Sounds good to me.

**Comment by noroadsleft** [comment]
Thanks!

For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

If you need any help with this just ask.

**Pull request merged**